dmtxdecodescheme.c

```diff
@@ -196,12 +206,13 @@ PushOutputMacroTrailer(DmtxMessage *msg)
  * \param  ptr
  * \param  dataEnd
  * \return Pointer to next undecoded codeword
+ *         NULL if an error was detected in the stream
  */
 static unsigned char *
 DecodeSchemeAscii(DmtxMessage *msg, unsigned char *ptr, unsigned char *dataEnd)
 {
    int upperShift;
-   int codeword, digits;
+   int codeword, digits, pushword;
```

Comment:
maybe `pushword` should be local to the `if` blocks where it's needed?

This makes the code easier to reason about by keeping all references to variables close to where they're used.

Reply:
I was keeping the coding style. I agree that locally is better. I have added a commit to do this for all variables in the function.

Reply:
Ah I hadn't appreciated that. You're right; it's better to keep with the current style.